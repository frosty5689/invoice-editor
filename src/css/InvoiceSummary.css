.Invoice-summary {
  display: flex;
  justify-content: flex-end;
}

.Invoice-summary-detail {
  display: flex;
  justify-content: space-between;
}

.Invoice-summary-detail-total {
  font-weight: bold;
}