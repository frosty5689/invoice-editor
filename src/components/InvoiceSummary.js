import React from 'react';
import { connect } from 'react-redux';
import { getLineItems } from '../redux/selectors';
import { CURRENCY_DECIMAL_PLACES, TAX_PERCENTAGE } from '../constants';
import '../css/InvoiceSummary.css';

export class InvoiceSummary extends React.Component  {
  getSubtotal(items) {
    const subtotal = items.reduce((sum, item) => { return sum + item.content.price * item.content.quantity }, 0);
    return subtotal;
  }
  getTax(amount) {
    return amount * TAX_PERCENTAGE;
  }
  toCurrency(number) {
    return number.toFixed(CURRENCY_DECIMAL_PLACES);
  }
  render() {
    const lineItems = this.props.lineItems
    const subtotal = this.getSubtotal(lineItems);
    const tax = this.getTax(subtotal);
    const total = subtotal + tax;

    return (
      <div className="Invoice-summary">
        <div>
          <div className="Invoice-summary-detail">
            <span>Subtotal</span>
            <span>&emsp;&emsp;</span>
            <span>${this.toCurrency(subtotal)}</span>
          </div>
          <div className="Invoice-summary-detail">
            <span>Tax (5%)</span>
            <span>&emsp;&emsp;</span>
            <span>${this.toCurrency(tax)}</span>
          </div>
          <div className="Invoice-summary-detail Invoice-summary-detail-total">
            <span>Total</span>
            <span>&emsp;&emsp;</span>
            <span>${this.toCurrency(total)}</span>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const lineItems = getLineItems(state);
  return { lineItems };
}

export default connect(mapStateToProps)(InvoiceSummary);