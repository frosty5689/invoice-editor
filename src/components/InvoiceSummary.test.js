import React from 'react';
import { InvoiceSummary } from './InvoiceSummary';
import renderer from 'react-test-renderer';

describe('InvoiceSummary component', () => {

  it('InvoiceSummary should render correctly', () => {
    const lineItems = [{
      id: 1,
      content: {
        name: 'Widget',
        quantity: 2,
        price: 10.22
      }
    }, {
      id: 2,
      content: {
        name: 'Cogs',
        quantity: 4,
        price: 2.22
      }
    }];

    const component = renderer.create(
      <InvoiceSummary lineItems={lineItems} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('InvoiceSummary getSubTotal should return correct subtotal', () => {
    const lineItems = [{
      id: 1,
      content: {
        name: 'Widget',
        quantity: 2,
        price: 10.22
      }
    }, {
      id: 2,
      content: {
        name: 'Cogs',
        quantity: 4,
        price: 2.22
      }
    }];
    const subtotal = InvoiceSummary.prototype.getSubtotal(lineItems);
    expect(subtotal).toBe(29.32);
  });

  it('InvoiceSummary getTax should return correct tax', () => {
    const tax = InvoiceSummary.prototype.getTax(100.00);
    expect(tax).toBe(5);
  });

  it('InvoiceSummary toCurrency should return two decimal places', () => {
    const currency = InvoiceSummary.prototype.toCurrency(100);
    expect(currency).toBe('100.00');
  });
});