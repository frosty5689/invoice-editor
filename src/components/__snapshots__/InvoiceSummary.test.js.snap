// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceSummary component InvoiceSummary should render correctly 1`] = `
<div
  className="Invoice-summary"
>
  <div>
    <div
      className="Invoice-summary-detail"
    >
      <span>
        Subtotal
      </span>
      <span>
          
      </span>
      <span>
        $
        29.32
      </span>
    </div>
    <div
      className="Invoice-summary-detail"
    >
      <span>
        Tax (5%)
      </span>
      <span>
          
      </span>
      <span>
        $
        1.47
      </span>
    </div>
    <div
      className="Invoice-summary-detail Invoice-summary-detail-total"
    >
      <span>
        Total
      </span>
      <span>
          
      </span>
      <span>
        $
        30.79
      </span>
    </div>
  </div>
</div>
`;
